specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+of their app type. This includes unpackaged apps and packaged win32 (MSIX Desktop Bridge, MSIX
+Win32App, Sparse Signed Packages). Moreover, all scenarios should adhere to OS resource
+management policies like Power Saver, Network Attribution (amount of data an app uses), enterprise
+group policies, etc. The WinAppSDK will abstract away the complexities of
+dealing with Notification delivery and related activations as much as possible freeing the developer
+to focus on other app related challenges.
+
+The App Notification API does the following:
+
+-   Allow Unpackaged apps to register for notifications.
+-   Allow Packaged apps to register for notifications.
+-   Allow Unpackaged apps to display cloud sourced notifications.
+-   Allow packaged apps to display cloud sourced notifications.
+-   Allow packaged apps to display local notifications
+-   Allow unpackaged apps to display local notifications
+-   Allow packaged apps to Update/Delete their local notifications.
+-   Allow unpackaged apps to Update/Delete their local notifications.
+-   Allow packaged applications to process notification activators.
+-   Allow unpackaged apps to process notification activators.
+
+# Examples
+
+## Packaged App Registering and handling a notification
+
+This scenario is specifically geared towards Packaged apps that need to send an app notification. An app
+Developer Registers their app for notifications and sets it up for background\foreground
+activation. Adding a AppNotificationActivated event handler signals that the app is ready to accept and handle
+an action in the foreground where the user clicks on a Notification in the action centre or a UI element in
+the payload body (like a Button). The developer can inturn run some business logic in their app in
+response to the UI Action.
+
+```cpp
+
+CLSID const c_comServerId = { 0x43136EB5,0xD36C,0x11CF,{0xAD,0xBC,0x00,0xAA,0x00,0xA8,0x00,0x33} };
+
+int main()
+{
+    AppNotificationActivationInfo activationInfo{ winrt::guid(c_comServerId) };
+    AppNotificationManager::Default().RegisterActivator(activationInfo);
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+
+    if (kind == ExtendedActivationKind::Launch)
+    {
+        // App is launched in Foreground. So intercept toast activators via Foreground event
+        const auto token = AppNotificationManager::Default().AppNotificationActivated([](const auto&, const AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+        {
+            ProcessNotificationArgs(notificationActivatedEventArgs);
+        });
+
+        // App does Foreground Stuff Here
+
+        // Cleanup
+        AppNotificationManager::Default().AppNotificationActivated(token);
+    }
+    else if (kind == ExtendedActivationKind::AppNotification)
+    {
+        auto notificationActivatedEventArgs = args.Data().as<AppNotificationActivatedEventArgs>();
+        ProcessNotificationArgs(notificationActivatedEventArgs);
```

Comment:
Why don't background notifications have to register for the AppNotificationActivated event?

Reply:
DISCUSSION: Raymond didn't undersatnd the API when he wrote this.